01-rstudio-intro.Rmd

```diff
@@ -378,10 +408,26 @@ rm(list <- ls())
 >
 
 
+#### R Packages
+
+It is possible to add functions to R by writing a package, or by
+obtaining a package written by someone else. As of this writing, there
+are over 7,000 packages available on CRAN (the comprehensive R archive
+network). R and RStudio have functionality for managing packages:
+
+* You can see what packages are installed by typing
+  `installed.packages()`
+* You can install packages by typing `install.packages("packagename")`,
+  where `packagename` is the package name, in quotes.
+* You can update installed packages by typing `update.packages()`
+* You can remove a package with `remove.packages("packagename")`
+* You can make a package available for use with `library(packagename)`
+
+
```

Comment:
On Tue, Aug 18, 2015 at 9:45 PM, Tom Wright <notifications@github.com>
wrote:

> In 01-rstudio-intro.Rmd
> <https://github.com/swcarpentry/r-novice-gapminder/pull/56#discussion_r37375430>
> :
>
> > +#### R Packages
> > +
> > +It is possible to add functions to R by writing a package, or by
> > +obtaining a package written by someone else. As of this writing, there
> > +are over 7,000 packages available on CRAN (the comprehensive R archive
> > +network). R and RStudio have functionality for managing packages:
> > +
> > +* You can see what packages are installed by typing
> > +  `installed.packages()`
> > +* You can install packages by typing `install.packages("packagename")`,
> > +  where `packagename` is the package name, in quotes.
> > +* You can update installed packages by typing `update.packages()`
> > +* You can remove a package with `remove.packages("packagename")`
> > +* You can make a package available for use with `library(packagename)`
> > +
> > +
>
> I like this description but as it stands it seems a bit theoretical for
> novices, I wonder if it just adds more cognitive load. I think it would be
> nicer with an example, my favourite 'toy' package is 'magic'.
>
>
> Yes, this is terse. My thought here is to have the reader understand 1)
you can install packages, 2) you can manage packages, 3) once you realize
you want to do this, you can return to this list. There is a challenge
asking them to install 3 packages.

This is my first pull request to SWC, so: What is the ideal procedure? Does
it make sense for me to try to respond to these comments, or is it better
for someone to accept the pull request and then someone else can edit
further? I'd be happy to add an install example, although I think that in a
class, the install challenge would create adequate opportunity for
experimentation and discussion.